app/src/views/private/components/drawer-item.vue

```diff
@@ -176,28 +176,25 @@ export default defineComponent({
 		);
 
 		const fields = computed(() => {
-			if (props.circularField) {
-				return fieldsWithPermissions.value.map((field: Field) => {
+			if (props.circularField || props.junctionField) {
+				return fieldsWithPermissions.value.filter((field: Field) => {
+					if (field.field === props.junctionField) {
+						set(field, 'meta.hidden', true);
+					}
 					if (field.field === props.circularField) {
+						if (props.primaryKey === '+') {
+							return false;
+						}
+
 						set(field, 'meta.readonly', true);
 					}
-					return field;
+					return true;
```

Comment:
Im not a big fan of the `filter` with side-effects but don't think extracting this into a `map`+`filter` would make it anymore readable